.circleci/cimodel/data/simple/bazel_definitions.py

```diff
@@ -0,0 +1,67 @@
+DOCKER_IMAGE = "/".join([
+    "308535385114.dkr.ecr.us-east-1.amazonaws.com",
+    "pytorch",
+    "pytorch-linux-xenial-py3.6-gcc7:f990c76a-a798-42bb-852f-5be5006f8026",
+])
+
+
+def gen_job_name(phase):
+    job_name_parts = [
+        "pytorch",
+        "bazel",
+        phase,
+    ]
+
+    return "_".join(job_name_parts)
+
+
+class BazelJob:
+    def __init__(self, phase, extra_props=None):
+        self.phase = phase
+        self.extra_props = extra_props or {}
+
+    def gen_tree(self):
+
+        template_parts = [
+            "pytorch",
+            "linux",
+            "bazel",
+            self.phase,
+        ]
+
+        build_env_parts = [
+            "pytorch",
+            "linux",
+            "xenial",
+            "py3.6",
+            "gcc7",
+            "bazel",
+            self.phase,
+        ]
+
+        full_job_name = gen_job_name(self.phase)
+        build_env_name = "-".join(build_env_parts)
+
+        extra_requires = [gen_job_name("build")] if self.phase == "test" else []
```

Comment:
Hmm, wouldn't it be more readable, if `extra_job_requires` were to put as an argument to `init` and then compute here as 
```suggestion
        extra_requires = [gen_job_name(job_name)] for job_name in self.extra_requires]
```